src/tree/revisionManagement/RevisionDraftItem.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode, Revision } from "@azure/arm-appcontainers";
+import { TreeElementBase, nonNullProp } from "@microsoft/vscode-azext-utils";
+import type { AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../utils/localize";
+import { treeUtils } from "../../utils/treeUtils";
+import type { ContainerAppModel } from "../ContainerAppItem";
+import { ScaleItem } from "../scaling/ScaleItem";
+import type { RevisionsItemModel } from "./RevisionItem";
+
+export class RevisionDraftItem implements RevisionsItemModel {
+    static readonly idSuffix: string = 'revisionDraft';
+    static readonly contextValue: string = 'revisionDraftItem';
+    static readonly contextValueRegExp: RegExp = new RegExp(RevisionDraftItem.contextValue);
+
+    id: string;
+    revisionsMode: KnownActiveRevisionsMode;
+
+    constructor(readonly subscription: AzureSubscription, readonly containerApp: ContainerAppModel, readonly revision: Revision) {
+        this.id = `${this.containerApp.id}/${RevisionDraftItem.idSuffix}`;
+        this.revisionsMode = containerApp.revisionsMode;
+    }
+
+    get revisionName(): string {
+        return nonNullProp(this.revision, 'name');
+    }
+
+    /**
+     * @example gets "rev-1" of "my-app--rev-1"
+     */
+    private get baseRevisionName(): string {
+        return this.revisionName.substring(this.revisionName.lastIndexOf('--') + 2);
+    }
+
+    static hasDescendant(item: RevisionsItemModel): boolean {
+        const revisionDraftBaseName: string | undefined = ext.revisionDraftFileSystem.getRevisionDraftFile(item)?.baseRevisionName;
+        return item.revision.name === revisionDraftBaseName;
+    }
+
+    getTreeItem(): TreeItem {
+        return {
+            id: this.id,
+            label: localize('draft', 'Draft'),
+            iconPath: treeUtils.getIconPath('revision-draft'),
+            description: this.containerApp.latestRevisionName === this.revisionName ?
+                localize('basedOnLatestRevision', 'Based on "{0}" (Latest)', this.baseRevisionName) :
+                localize('basedOnRevision', 'Based on "{0}"', this.baseRevisionName),
+            contextValue: RevisionDraftItem.contextValue,
+            collapsibleState: TreeItemCollapsibleState.Expanded
+        };
+    }
+
+    getChildren(): TreeElementBase[] {
```

Comment:
This being the same as the `ContainerAppItem` kind of makes me want this class to be extended from or based off of a an abstract `ContainerAppItemBase`. I think it'll be kind of annoying when we change/add children or the container app item itself.

Reply:
Actually, instead of enforcing inheritance for this, how do you feel about providing the children through a static method on the `RevisionItem`.  That feels simpler to me and we can still manage the children in one place.  It also makes sense to keep it on that item because the template changes are analogous to the concept of revisions.